src/Merchants.php

```diff
@@ -127,13 +127,14 @@ private static function get_merchant_id_from_advertiser() {
 	 */
 	public static function update_or_create_merchant() {
 
-		$local_feed = ProductFeedStatus::get_local_feed();
+		$configs = LocalFeedConfigs::get_instance()->get_configurations();
+		$config  = reset( $configs );
 
 		$merchant_name = apply_filters( 'pinterest_for_woocommerce_default_merchant_name', esc_html__( 'Auto-created by Pinterest for WooCommerce', 'pinterest-for-woocommerce' ) );
 
 		$args = array(
 			'merchant_domains' => get_home_url(),
-			'feed_location'    => $local_feed['feed_url'],
+			'feed_location'    => $config['feed_url'],
```

Comment:
What happens when for some reason there is no data returned and this is unset, or `feed_url` doesn't exist?

Reply:
We always have configs, at least one. The LocalFeedConfigs is created this way. There feed file may not be generated yet and there can be nothing to fetch under `feed_url` but the URL already exists in the config.